content/zh-cn/docs/reference/kubectl/generated/kubectl_set/kubectl_set_selector.md

```diff
@@ -0,0 +1,658 @@
+-->
+针对某个资源设置选择算符。请注意，
+如果资源在 `set selector` 调用之前已有选择算符，则新的选择算符将覆盖旧的选择算符。
+
+选择算符必须以字母或数字开头，可以包含字母、数字、连字符、点和下划线，最长为 63 个字符。
+如果指定了 --resource-version，则更新将使用此资源版本，否则将使用现有的资源版本。
+注意：目前选择算符只能被设置在 Service 对象上。
+
+```shell
+kubectl set selector (-f FILENAME | TYPE NAME) EXPRESSIONS [--resource-version=version]
+```
+
+## {{% heading "examples" %}}
+
+<!--
+```
+  # Set the labels and selector before creating a deployment/service pair
+  kubectl create service clusterip my-svc --clusterip="None" -o yaml --dry-run=client | kubectl set selector --local -f - 'environment=qa' -o yaml | kubectl create -f -
+  kubectl create deployment my-dep -o yaml --dry-run=client | kubectl label --local -f - environment=qa -o yaml | kubectl create -f -
+```
+-->
+```shell
+# 在创建 Deployment/Service 对之前设置标签和选择算符
+kubectl create service clusterip my-svc --clusterip="None" -o yaml --dry-run=client | kubectl set selector --local -f - 'environment=qa' -o yaml | kubectl create -f -
+kubectl create deployment my-dep -o yaml --dry-run=client | kubectl label --local -f - environment=qa -o yaml | kubectl create -f -
+```
+
+## {{% heading "options" %}}
+
+   <table style="width: 100%; table-layout: fixed;">
+<colgroup>
+<col span="1" style="width: 10px;" />
+<col span="1" />
+</colgroup>
+<tbody>
+
+<tr>
+<td colspan="2">--all</td>
+</tr>
+<tr>
+<td></td><td style="line-height: 130%; word-wrap: break-word;"><p>
+<!--
+Select all resources in the namespace of the specified resource types
+-->
+选择指定资源类型的命名空间中的所有资源。
+</p></td>
+</tr>
+
+<tr>
+<td colspan="2">--allow-missing-template-keys&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<!--Default:-->默认值：true</td>
+</tr>
+<tr>
+<td></td><td style="line-height: 130%; word-wrap: break-word;"><p>
+<!--
+If true, ignore any errors in templates when a field or map key is missing in the template. Only applies to golang and jsonpath output formats.
+-->
+如果为 true，在模板中字段或映射键缺失时忽略模板中的错误。
+仅适用于 golang 和 jsonpath 输出格式。
+</p></td>
+</tr>
+
+<tr>
+<td colspan="2">--dry-run string[="unchanged"]&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<!--Default:-->默认值："none"</td>
+</tr>
+<tr>
+<td></td><td style="line-height: 130%; word-wrap: break-word;"><p>
+<!--
+Must be &quot;none&quot;, &quot;server&quot;, or &quot;client&quot;. If client strategy, only print the object that would be sent, without sending it. If server strategy, submit server-side request without persisting the resource.
+-->
+必须是 "none"、"server" 或 "client"。如果是 client 策略，仅打印将要发送的对象，而不实际发送。
+如果是 server 策略，提交服务器端请求而不持久化资源。
+</p></td>
+</tr>
+
+<tr>
+<td colspan="2">--field-manager string&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<!--Default:-->默认值："kubectl-set"</td>
+</tr>
+<tr>
+<td></td><td style="line-height: 130%; word-wrap: break-word;"><p>
+<!--
+Name of the manager used to track field ownership.
+-->
+用于跟踪字段属主关系的管理器的名称。
+</p></td>
+</tr>
+
+<tr>
+<td colspan="2">-f, --filename strings</td>
+</tr>
+<tr>
+<td></td><td style="line-height: 130%; word-wrap: break-word;"><p>
+<!--
+identifying the resource.
```

Comment:
这个不太理解，为什么用文件名，而且是 strings（不是 string）来标识资源？
请详查。

Reply:
从 k/k 只找到[这里](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cli-runtime/pkg/genericclioptions/builder_flags.go)有 2个 identifying the resource.

```go
// NewResourceBuilderFlags returns a default ResourceBuilderFlags
func NewResourceBuilderFlags() *ResourceBuilderFlags {
	filenames := []string{}

	return &ResourceBuilderFlags{
		FileNameFlags: &FileNameFlags{
			Usage:     "identifying the resource.",
			Filenames: &filenames,
			Recursive: boolPtr(true),
		},
	}
}

// WithFile sets the FileNameFlags.
// If recurse is set, it will process directory recursively. Useful when you want to manage related manifests
// organized within the same directory.
func (o *ResourceBuilderFlags) WithFile(recurse bool, files ...string) *ResourceBuilderFlags {
	o.FileNameFlags = &FileNameFlags{
		Usage:     "identifying the resource.",
		Filenames: &files,
		Recursive: boolPtr(recurse),
	}

	return o
}
```

Reply:
第二个函数的注释说明，这个 flag 的用途是用来在同一目录下存在多个文件时，使用文件名来区分不同的资源。